packages/adp-tooling/src/writer/index.ts

```diff
@@ -42,26 +59,41 @@ export async function generate(basePath: string, config: AdpWriterConfig, fs?: E
     if (!fs) {
         fs = create(createStorage());
     }
-    const tmplPath = join(__dirname, '../../templates/project');
     const fullConfig = setDefaults(config);
 
     fs.copyTpl(join(tmplPath, '**/*.*'), join(basePath), fullConfig, undefined, {
         globOptions: { dot: true },
         processDestinationPath: (filePath: string) => filePath.replace(/gitignore.tmpl/g, '.gitignore')
     });
 
-    // ui5.yaml
-    const ui5ConfigPath = join(basePath, 'ui5.yaml');
-    const baseUi5ConfigContent = fs.read(ui5ConfigPath);
-    const ui5Config = await UI5Config.newInstance(baseUi5ConfigContent);
-    enhanceUI5Yaml(ui5Config, fullConfig);
-    fs.write(ui5ConfigPath, ui5Config.toString());
-    // ui5-deploy.yaml
-    if (hasDeployConfig(fullConfig)) {
-        const ui5DeployConfig = await UI5Config.newInstance(baseUi5ConfigContent);
-        enhanceUI5DeployYaml(ui5DeployConfig, fullConfig);
-        fs.write(join(basePath, 'ui5-deploy.yaml'), ui5DeployConfig.toString());
+    await writeUi5Yaml(basePath, fullConfig, fs);
+
+    return fs;
+}
+
+/**
+ * Writes the adp-project template to the mem-fs-editor instance during migration.
+ *
+ * @param basePath - the base path
+ * @param config - the writer configuration
+ * @param fs - the memfs editor instance
+ * @returns the updated memfs editor instance
+ */
+
+export async function migrateAdp(basePath: string, config: AdpWriterConfig, fs?: Editor): Promise<Editor> {
```

Comment:
What is the difference between `migrateAdp` and `generate`? I understand what the different use cases are but I don't see the difference in the code.
Shouldn't you handle the possibility of an existing `ui5.yaml` or `package.json`?

Reply:
The idea is to extend it in the near future, exactly for what you said
>Shouldn't you handle the possibility of an existing ui5.yaml or package.json?